### DISCUSSION TITLE
How to add coverage for mutated RemoveMethodCall?

### DISCUSSION THREAD

**Discussion by wietsewarendorff** [pestphp/pest#1272] [Q&A]
I really like the mutation-testing in v3, but I have a situation. Consider the following pseudo-code:

```
// third party
function foo(?string $foo = null): ?string {
    return $foo;
}

// my code
function bar(string $bar): string
{
    $bar = foo($bar);
    assert(is_string($bar));

    return $bar;
}
```

To make sure `bar()` cannot return `null` I added an assertion (phpstan likes it better this way too). For me this is where assertions shine: `$bar` should always be a string in this case.

Now when I run a mutation test, pest removes the assertion and claims the line is not covered (which is true). However, there is no way to test this. Now I could add `// @pest-mutate-ignore` but then I have to add it to all assertions. And the assertion is not wrong here, it's just to make sure the code acts the way I expect it to do (for instance if the vendor changes the api of `foo()`).

Any suggestions, or am I missing something obvious here?

**Comment by akosglue** [comment]
can you share the test?

**Comment by akosglue** [comment]
normally you had something like this I guess:

```
test('assert_test', function () {
    $b = new Bar();
    $s = $b->bar('asd');
    $this->assertIsString($s);
    $this->assertEquals('asd', $s);
});
```

And then you need to add a second case I suppose:

```
test('assert_test_exception', function () {
    $this->expectException(AssertionError::class);
    $bar = new Bar();
    $s = $bar->bar(null);
    $this->assertIsString($s);
    $this->assertEquals('asd', $s);
});
```

But in this case you need to relax your code from this:

```
    public function bar(string $bar): string
    {
        $bar = $this->foo($bar);
        assert(is_string($bar));
        return $bar;
    }
```

to this:

```
    public function bar(?string $bar): string
    {
        $bar = $this->foo($bar);
        assert(is_string($bar));
        return $bar;
    }
```

Does that make sense?

**Comment by wietsewarendorff** [comment]
It does make sense... thanks for your answer, I will look into this and see if I can do it this way. Thanks!
